src/print.ml

```diff
@@ -27,43 +34,94 @@ let pp_meta : meta pp = fun oc m ->
 
 (** [pp_term oc t] prints the term [t] to the channel [oc]. *)
 let pp_term : term pp = fun oc t ->
-  let out oc fmt = Format.fprintf oc fmt in
-  (* NOTE we apply the conventions used in [Parser.expr] for priorities. *)
+  let out = Format.fprintf in
   let rec pp (p : [`Func | `Appl | `Atom]) oc t =
-    let pp_func = pp `Func in
-    let pp_appl = pp `Appl in
-    let pp_atom = pp `Atom in
+    let (h, args) = Basics.get_args t in
+    let args =
+      if !print_implicis then args else
+      let impl =  match h with Symb(s,_) -> s.sym_impl | _ -> [] in
+      let rec filter_impl impl args =
```

Comment:
Why not putting this generic function on lists in Extra like in my PR?

Reply:
We could, but it is unlikely that this function will be required somewhere else. And I think that it is easier to see what we are doing here with the implementation of the function next to its use. If you insist, I can move it! :)